setup-ruby-and-rust/action.yml

```diff
@@ -194,6 +193,7 @@ runs:
         path: |
           **/tmp/${{ steps.set-outputs.outputs.ruby-platform }}/
           ${{ inputs.cargo-cache-extra-path }}
+          ${{ env.RB_SYS_DOCK_CACHE_DIR }}
```

Comment:
Could this have broken caching?

I've been experimenting with speeding up wasmtime-rb's builds and can't get caching to work right now. Restoring the cache fails with a version mismatch (pardon the wall-of-text ouput):


>##[debug]No matching cache found for cache key 'v0__202301109074__x86_64-linux__00af9384e5cb8280eb12b02e39d14d4d234fa347da71f1761b25afba69c672b7__1c34b7cb87ce8c3b988ff34e736f39bbb89c38242774fd74fa27f37fb0667fb3__2ed5166006f90d23922b24a1ff788c511cf67c21457c97f4040b72980781818c', version 'baad146dba25e4bb6237818902b7dc126cdbd762cb7e424c45441c83c2b10a73 and scope refs/heads/faster-ci. There exist one or more cache(s) with similar key but they have different version or scope. See more info on cache matching here: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key 
##[debug]Other caches with similar key:
##[debug]Cache Key: v0__202301109074__x86_64-linux__00af9384e5cb8280eb12b02e39d14d4d234fa347da71f1761b25afba69c672b7__1c34b7cb87ce8c3b988ff34e736f39bbb89c38242774fd74fa27f37fb0667fb3__2ed5166006f90d23922b24a1ff788c511cf67c21457c97f4040b72980781818c, Cache Version: 89c19b399e6310f68e105803e9f98046f2013b609b7a50af9150fac3f7dc0391, Cache Scope: refs/heads/faster-ci, Cache Created: 2023-01-15T22:02:13.93Z

I think what's going on is that when first trying to restore the cache, `env.RB_SYS_DOCK_CACHE_DIR` is null, so the paths are `[tmp/$ruby-platform]`. When writing the cache at the end of the job, we've now set `env.RB_SYS_DOCK_CACHE_DIR`, so the paths are `[tmp/$ruby-platform, $HOME/.cache/rb-sys-dock]`. Makes sense?

Assuming I'm right, I guess a simple fix would be to move up the _Configure environment_ step.